src/core/server/http/lifecycle/auth.ts

```diff
@@ -48,8 +48,9 @@ const authResult = {
   authenticated(data: Partial<AuthResultData> = {}): AuthResult {
     return {
       type: ResultType.authenticated,
-      state: data.state || {},
-      headers: data.headers || {},
+      state: data.state,
+      headers: data.headers,
```

Comment:
@azasypkin are you okay if rename `headers` --> `requestHeaders`? I can change Security plugin code accordingly https://github.com/restrry/kibana/blob/fa48e98ed2dfb823a38988a7f7da6e8fbdd214b8/x-pack/plugins/security/server/authentication/index.ts#L96

Reply:
Yeah, that's fine by me